sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVPartitionReaderFactory.scala

```diff
@@ -58,7 +58,7 @@ case class CSVPartitionReaderFactory(
       actualReadDataSchema,
       options,
       filters)
-    val schema = if (options.columnPruning) actualReadDataSchema else actualDataSchema
+    val schema = if (options.isColumnPruningEnabled) actualReadDataSchema else actualDataSchema
```

Comment:
https://github.com/apache/spark/blob/829e742df8251c6f5e965cb08ad454ac3ee1a389/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala#L103

https://github.com/apache/spark/blob/829e742df8251c6f5e965cb08ad454ac3ee1a389/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala#L128

@MaxGekk Should the check in `CSVFileFormat` be changed to `parsedOptions.isColumnPruningEnabled` too?

Reply:
The `schema` is used only `CSVHeaderChecker` which is supposed to check column names in CSV and provided schema fields. It shouldn't depend on the column pruning feature at all, from my point of view.

```scala
  private def checkHeaderColumnNames(columnNames: Array[String]): Unit = {
...
      if (headerLen == schemaSize) {
...
      } else {
        errorMessage = Some(
          s"""|Number of column in CSV header is not equal to number of fields in the schema:
              | Header length: $headerLen, schema size: $schemaSize
              |$source""".stripMargin)
      }
```

`schemaSize` must be f**ull data schema** of CSV filed, but not the required schema.

Let me re-think it, and avoid the dependency from the column pruning in `CSVHeaderChecker`.

Reply:
@LuciferYang Actually, you are right. Please, review this follow up PR: https://github.com/apache/spark/pull/44910